src/main/groovy/life/qbic/datamodel/dtos/business/services/ExternalServiceProduct.groovy

```diff
@@ -23,7 +23,7 @@ class ExternalServiceProduct extends PartialProduct {
      * @param productId A product id, uniquely identifying the product in the offer environment
      * @param serviceProvider The service provider
      */
-    ExternalServiceProduct(String name, String description, double internalUnitPrice, double externalUnitPrice, ProductUnit unit, ProductId productId, Facility serviceProvider) {
-        super(name, description, internalUnitPrice, externalUnitPrice, unit, productId, serviceProvider)
+    ExternalServiceProduct(String name, String description, double internalUnitPrice, double externalUnitPrice, ProductUnit unit, long runningNumber, Facility serviceProvider) {
+        super(name, description, internalUnitPrice, externalUnitPrice, unit, new ProductId.Builder(ProductType.EXTERNAL_SERVICE.toString(), runningNumber).build(), serviceProvider)
```

Comment:
Maybe move the construction building out of this super constructor. It makes it hard to read. (sadly I cant make a suggestion, dont know why)

Reply:
```suggestion
       ProductId externalProductId = new ProductId.Builder(ProductType.EXTERNAL_SERVICE.toString(), runningNumber).build()
        super(name, description, internalUnitPrice, externalUnitPrice, unit, externalProductId, serviceProvider)
```